File: backend/app/api/schemas/attendance.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel


class AttendanceRecordResponse(BaseModel):
    date: date
    check_in: datetime | None = None
    check_out: datetime | None = None
    status: str  # "on_time" | "late" | "absent"


class AttendancePolicyResponse(BaseModel):
    checkin_expected: str
    checkout_expected: str
    late_threshold: str
    timezone: str


class RecruiterAttendanceRow(BaseModel):
    user_id: uuid.UUID
    first_name: str
    last_name: str
    employee_code: str | None = None
    designation: str | None = None
    check_in: datetime | None = None
    check_out: datetime | None = None
    status: str  # "on_time" | "late" | "absent"

File: frontend/src/api/attendance.ts
import apiClient from '@/api/client'

export interface AttendanceRecord {
  date: string
  check_in: string | null
  check_out: string | null
  status: 'on_time' | 'late' | 'absent'
}

export interface RecruiterAttendanceRow {
  user_id: string
  first_name: string
  last_name: string
  employee_code: string | null
  designation: string | null
  check_in: string | null
  check_out: string | null
  status: 'on_time' | 'late' | 'absent'
}

export async function fetchMyToday(): Promise<AttendanceRecord> {
  const { data } = await apiClient.get<AttendanceRecord>('/api/v1/attendance/today/me')
  return data
}

export async function checkIn(): Promise<AttendanceRecord> {
  const { data } = await apiClient.post<AttendanceRecord>('/api/v1/attendance/checkin')
  return data
}

export async function checkOut(): Promise<AttendanceRecord> {
  const { data } = await apiClient.post<AttendanceRecord>('/api/v1/attendance/checkout')
  return data
}

export async function fetchAllToday(): Promise<RecruiterAttendanceRow[]> {
  const { data } = await apiClient.get<RecruiterAttendanceRow[]>('/api/v1/attendance/today')
  return data
}

export interface AttendancePolicy {
  checkin_expected: string
  checkout_expected: string
  late_threshold: string
  timezone: string
}

export async function fetchPolicy(): Promise<AttendancePolicy> {
  const { data } = await apiClient.get<AttendancePolicy>('/api/v1/attendance/policy')
  return data
}
